actions/workflows/st2_chatops_e2e_tests.yaml

```diff
@@ -0,0 +1,110 @@
+---
+version: '1.0'
+
+input:
+  - cwd
+
+tasks:
+  skip_if_ubuntu_14:
+    action: core.local
+    input:
+      command: "lsb_release --release | awk '{ print $2 }'"
+    next:
+      - when: <% succeeded() and (result().stdout != '14.04') %>
+        publish:
+          - websocket:
+        do:
+          - setup
+      - when: <% failed() %>
+        publish:
+          - ubuntu_version: <% result().stdout %>
+        do:
+          - log_ubuntu_version
+          - skip_if_rhel_6
+  log_ubuntu_version:
+    action: core.echo
+    input:
+      message: <% ctx().ubuntu_version %>
+  skip_if_rhel_6:
+    action: core.local
+    input:
+      command: "[[ -e /etc/redhat-release ]] && cat /etc/redhat-release 2>/dev/null | sed -r 's/^[a-zA-Z ]*([0-9]){1,}\..*$/\\1/'"
+    next:
+      - when: <% succeeded() and (int(result().stdout) >= 7) %>
+        publish:
+          - websocket: /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem
+        do:
+          - setup
+      - when: <% failed() %>
+        publish:
+          - redhat_version: <% result().stdout %>
+        do:
+          - log_redhat_version
+          - noop
+  log_redhat_version:
+    action: core.echo
+    input:
+      message: <% ctx().redhat_version %>
+
+  # Setup the virtualenv
+  # This is separated from the configure action because it does not (and SHOULD
+  # not be) run with root privileges
+  setup:
+    action: st2cd.st2_chatops_setup_e2e_tests
+    input:
+      cwd: <% ctx().cwd %>
+    next:
+      - when: <% succeeded() %>
+        do:
+          - run_with_slack
```

Comment:
So it looks like we can't run those 3 different adapter tests in parallel?

Reply:
The MS Teams and Mattermost tasks are just `core.noop` actions, so I don't think they are going to slow the workflow down any appreciable amount.

Since I haven't written the end-to-end tests for the other adapters, I'm also not sure if they can be parallelized yet, so I figured the placeholders can run in series.

I absolutely will keep parallelization of the distinct end-to-end adapter tests in mind when I am writing them.